akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -25,5 +25,12 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val addressUid: Int = ThreadLocalRandom.current.nextInt()
-}
\ No newline at end of file
+  val longAddressUid: Long = {
+    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
```

Comment:
My point is that we can't use different uid for cluster and remoting, becuase that uid is used by clustering when it quarantines a removed node and then it must be the full length uid if that is what is used in remoting. But it might be possible that we use a full length uid under the hood in clustering, but only expose the low 32 for users (in existing api, such as `Member`)